src/actions/users.js

```diff
@@ -29,14 +31,52 @@ const getUsers = async () => async (dispatch, getState) => {
       });
     }
 
-    const url = `//${apiHost + SPACES_ENDPOINT}/${spaceId}/${USERS_ENDPOINT}`;
+    // we have two endpoints for users
+    const spaceUsersUrl = `//${
+      apiHost + SPACES_ENDPOINT
+    }/${spaceId}/${USERS_ENDPOINT}`;
+    const graaspUsersUrl = `//${apiHost + GRAASP_USERS_ENDPOINT}`;
 
-    const response = await fetch(url, DEFAULT_GET_REQUEST);
+    // get users that are associated with this space
+    const spaceUsersResponse = await fetch(spaceUsersUrl, DEFAULT_GET_REQUEST);
 
     // throws if it is an error
-    await isErrorResponse(response);
+    await isErrorResponse(spaceUsersResponse);
+
+    const spaceUsers = await spaceUsersResponse.json();
+
+    // complement with graasp users that interacted with the application
+    // these are users that access the app using their credentials
+    // through the page view interface
+    const {
+      appInstanceResources: { content },
+    } = getState();
+    const graaspUsers = [];
+    // eslint-disable-next-line no-restricted-syntax
+    for (const appInstanceResourceUser of content) {
+      const userId = appInstanceResourceUser.user;
+      if (!spaceUsers.map((user) => user.id).includes(userId)) {
```

Comment:
It might worth saving `spaceUsers.map((user) => user.id)` in a variable before the loop in order to not compute it each time.

Reply:
Good catch.